src/ploigos_step_runner/utils/xml.py

```diff
@@ -87,3 +91,65 @@ def get_xml_element_by_path(xml_file_path, xpath, default_namespace=None, xml_na
         return xml_file.find(xpath)
 
     return xml_file.find(xpath, namespaces)
+
+def aggregate_xml_element_attribute_values(xml_file_path, xml_element, attribs):
+    """
+    Function to parse a XML file looking for a specific element and obtaining its attributes.
+
+    If element is not found in file the file is ignored.
+
+    If attribute is not found in element the file is ignored.
+
+    Parameters
+    ----------
+    xml_file_path: str
+        Path to directory containing XML files
+    xml_element: str
+        XML element being searched for
+    attribs: list
+        List of attributes being searched for in the XML element
+
+    Returns:
+
+    report_results: dict
+        A dictionary of attribute values.
+
+    Raises
+    ------
+    ValueError
+        If XML element is not found in file(s)
+
+    """
+
+    report_results = {}
+
+    #Search file path for XML files
+
+    if os.path.isdir(xml_file_path):
+        xml_files = glob.glob(xml_file_path + '/*.xml', recursive=False)
+    elif os.path.isfile(xml_file_path):
+        xml_files = [xml_file_path]
+
+    with open(xml_files[0], "r") as xml_file:
+        print("line: " + xml_file.read())
+
+    #Iterate over XML files
+    for file in xml_files:
+        try:
+            pom_element = get_xml_element(file, xml_element)
+            #Add attributes to dictionary
+            for attrib in attribs:
+                if attrib in pom_element.attrib:
+                    attribute_value = pom_element.attrib[attrib]
+                    # aggregate attribute accross multiple files
+                    if attrib in report_results:
+                        report_results[attrib] = str(float(report_results[attrib])
```

Comment:
why convert to a string?

what if the value we want isn't a number? i see this function being used in future to get attribute values that may not be numbers, can add condition for value not being a float?

Reply:
If the value is a string should we concatenate the values instead? Maybe with commas?

Reply:
Since the get_xml_element always returns Strings I might have to check with isnumeric()

Reply:
Actually an easy way would be to try/except converting them to floats and if that excepts then just treat them as strings.

Reply:
What happens in a case where the element in some files is a number and in others a string? We wouldn't be able to go back and re-stringify the numbers we've already added together. Would it be better to just store all of the values, and then after parsing all files, check if they are all numeric, then add?

Reply:
Yeah I think that would make sense. It will have to iterate over the list for each attribute at least once to verify if they are floatable or only stringable and then iterate to either sum them or concatenate them.

Reply:
I have also added a check for if it's an integer and not a float so we don't have something like tests = 42.0 and instead tests = 42.